Skip files and non-chart folders

We have a folder in charts folder that is not a chart but is related to charts that this action fails on. We also have a README.md in the charts folder that causes it to fail. It would be nice if the action could skip files and directories that don't have a Chart.yaml.

Also, it should run helm dep up before lint since parent chart may use child values to render templates and fail if it doesn't have those charts downloaded yet.

Or linting should be optional, as we do it in a previous step anyway.

```
Error: 7 chart(s) linted, 3 chart(s) failed
==> Linting /github/workspace/charts/README.md
Error unable to check Chart.yaml file in chart: stat /github/workspace/charts/README.md/Chart.yaml: not a directory

==> Linting /github/workspace/charts/hedera-mirror

==> Linting /github/workspace/charts/hedera-mirror-common
[ERROR] templates/: template: hedera-mirror-common/templates/NOTES.txt:8:45: executing "hedera-mirror-common/templates/NOTES.txt" at <.Values.traefik.ports.web.exposedPort>: nil pointer evaluating interface {}.exposedPort

==> Linting /github/workspace/charts/hedera-mirror-grpc

==> Linting /github/workspace/charts/hedera-mirror-importer

==> Linting /github/workspace/charts/hedera-mirror-rest

==> Linting /github/workspace/charts/marketplace
Error unable to check Chart.yaml file in chart: stat /github/workspace/charts/marketplace/Chart.yaml: no such file or directory
```

--- a/src/entrypoint.sh
+++ b/src/entrypoint.sh
@@ -11,6 +11,7 @@ REPOSITORY=$5
 BRANCH=$6
 HELM_VERSION=$7
 
+CHARTS=()
 CHARTS_TMP_DIR=$(mktemp -d)
 REPO_ROOT=$(git rev-parse --show-toplevel)
 REPO_URL=""
@@ -44,10 +45,23 @@ main() {
       REPO_URL="https://x-access-token:${GITHUB_TOKEN}@github.com/${OWNER}/${REPOSITORY}"
   fi
 
+  locate
   download
+  dependencies
   lint
   package
   upload
+}
+
+locate() {
+  for dir in $(find "${CHARTS_DIR}" -type d -mindepth 1 -maxdepth 1); do
+    if [[ -f "${dir}/Chart.yaml" ]]; then
+      CHARTS+=("${dir}")
+      echo "Found chart directory ${dir}"
+    else
+      echo "Ignoring non-chart directory ${dir}"
+    fi
+  done
 }
 
 download() {
@@ -62,12 +76,18 @@ download() {
   rm -rf $tmpDir
 }
 
+dependencies() {
+  for chart in ${CHARTS[@]}; do
+    helm dependency update "${chart}"
+  done
+}
+
 lint() {
-  helm lint ${REPO_ROOT}/${CHARTS_DIR}/*/
+  helm lint ${CHARTS[*]}
 }
 
 package() {
-  helm package -u ${REPO_ROOT}/${CHARTS_DIR}/*/ --destination ${CHARTS_TMP_DIR}
+  helm package ${CHARTS[*]} --destination ${CHARTS_TMP_DIR}
 }
 
 upload() {

--- a/testdata/notchart/README.md
+++ b/testdata/notchart/README.md
@@ -1,0 +1,1 @@
+This folder is not a chart and should be skipped

